sdk/formrecognizer/azure-ai-formrecognizer/samples/async_samples/sample_differentiate_custom_forms_with_labeled_and_unlabeled_models_async.py

```diff
@@ -0,0 +1,97 @@
+# coding: utf-8
+
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+"""
+FILE: sample_differentiate_custom_forms_with_labeled_and_unlabeled_models_async.py
+
+DESCRIPTION:
+    This sample demonstrates the differences in output that arise when recognize_custom_forms
+    is called with custom models trained with labeled and unlabeled data.
+USAGE:
+    python sample_differentiate_custom_forms_with_labeled_and_unlabeled_models_async.py
+
+    Set the environment variables with your own values before running the sample:
+    1) AZURE_FORM_RECOGNIZER_ENDPOINT - the endpoint to your Cognitive Services resource.
+    2) AZURE_FORM_RECOGNIZER_KEY - your Form Recognizer subscription key
+    3) CUSTOM_TRAINED_LABELED_MODEL_ID - the ID of your custom model trained with labeled data
+    4) CUSTOM_TRAINED_UNLABELED_MODEL_ID - the ID of your custom model trained with unlabeled data
+"""
+
+import os
+import asyncio
+
+
+class DifferentiateCustomFormsWithLabeledAndUnlabeledModelsAsync(object):
+
+    endpoint = os.environ["AZURE_FORM_RECOGNIZER_ENDPOINT"]
+    key = os.environ["AZURE_FORM_RECOGNIZER_KEY"]
+    labeled_model_id = os.environ["CUSTOM_TRAINED_LABELED_MODEL_ID"]
+    unlabeled_model_id = os.environ["CUSTOM_TRAINED_UNLABELED_MODEL_ID"]
+
+    async def recognize_custom_forms(self):
+        from azure.core.credentials import AzureKeyCredential
+        from azure.ai.formrecognizer.aio import FormRecognizerClient
+        async with FormRecognizerClient(
+            endpoint=self.endpoint, credential=AzureKeyCredential(self.key)
+        ) as form_recognizer_client:
+
+            # The form you are recognizing must be of the same type as the forms the custom model was trained on
+            with open("sample_forms/forms/Form_1.jpg", "rb") as f:
+                stream = f.read()
+            forms_with_labeled_model = await form_recognizer_client.recognize_custom_forms(
+                model_id=self.labeled_model_id, stream=stream
+            )
+            forms_with_unlabeled_model = await form_recognizer_client.recognize_custom_forms(
+                model_id=self.unlabeled_model_id, stream=stream
+            )
+
+            # The main difference is found in the labels of its fields
+            # The form recognized with a labeled model will have the labels it was trained with,
```

Comment:
So, this is a suboptimal aspect of the current API design, but we use "label" to mean two different things right now.  There are training-time labels that are designated in the label files.  There are also form field labels that are the titles of the form fields in the form (for example, [this](https://uxmovement.com/wp-content/uploads/2012/03/right-vs-left-form-labels.png)).  `field.Name` uniquely identifies the form field (and is populated from training-time labels), where as `field.Label` is the form field label.

Reply:
In Python, we return fields as a dictionary, so I never really access field.name in the sample (would be too clunky), so I don't think the best way to clarify the comment would be to address them as field.name. Let me know if my explanation makes sense!

Reply:
If I understand correctly, let's rename `label` to `name` in nested for statement:

```
for labeled_form in forms_with_labeled_model:
    for **name**, field in labeled_form.fields.items():
```

Because although we're using the var name `label` in the loop, this is actually represented by the `name` attribute in FormField. We might also want to do this in unlabeled below, as `name` really holds the unique name of the field. Make sense? ping me if not